mne/stats/cluster_level.py

```diff
@@ -1729,3 +1737,310 @@ def summarize_clusters_stc(
     data_summary[:, 0] = np.sum(data_summary, axis=1)
 
     return klass(data_summary, vertices, tmin, tstep, subject)
+
+
+def cluster_test(
+    df: pd.DataFrame,
+    formula: str = None,  # Wilkinson notation formula for design matrix
+    n_permutations: int = 10000,
+    seed: None | int | np.random.RandomState = None,
+    tail: int = 0,  # 0 for two-tailed, 1 for greater, -1 for less
+    n_jobs: int = 1,  # how many cores to use
+    adjacency: tuple = None,
```

Comment:
the annotation is incorrect, must be tuple | None

you will need to 

from __future__ import annotations

to make it work with Py3.9 (see one of my recent PRs)